design/manual-authn-and-authz/README.md

```diff
@@ -0,0 +1,65 @@
+# Manual Authentication and Authorization workaround
+
+* Status: **Proposal**
+* Discussion: [GitHub PR](https://github.com/strimzi/strimzi-kafka-operator/pull/599)
+
+## Motivation
+
+The long term plan for Strimzi is to have custom resources to define Users for Authentication and Authorization rules.
+The custom resources should be read by operator style application which would be responsible for the setup of the users or ACL rules.
+Until such operator (User Operator - UO) is available, it would be useful to have a way how to setup Authentication and Authorization at least in manual / semi-manual way.
+This proposal covers such workaround.
+There is no long term commitment to support the mechanisms suggested in this proposal.
+After the final implementation of the operator for Authorization and Authentication is implemented, it will not be possible anymore to use this workaround.
+
+## Authentication
+
+The Cluster Operator (CO) maintains a _Clients CA_. 
+This Client CA is set as truststore for the TLS enabled Kafka interface which is designated for client connections (port 9093).
+Currently, this _Clients CA_ does not enforce client authentication.
+And enabling such configuration through `spec.kafka.config` is not possible because the required options are in the _forbidden options_ list.
+This proposal suggests to modify the list of forbidden options to allow end-users to enable or disable the TLS client authentication.
+This change will be only temporary and will be removed once the actual UO is in place with the needed configuration.
+
+User who want to use manual setup of authentication will be able to do so in the following steps:
+
+* Modify the Kafka cluster resource and add configuration options to enable TLS client authentication for the clients port (option `listener.name.clienttls.ssl.client.auth=required`)
+* Use the Clients CA to manually issue user certificates. 
+  The CA public and private key can be downloaded from the Kubernetes / OpenShift secret and the user certificates can be signed separately. 
+  The CA private key should be kept secret.
+* The authenticated users will be given the Principal name based on the subject of their certificate (e.g. `CN=writeuser,OU=Unknown,O=Unknown,L=Unknown,ST=Unknown,C=Unknown`)
+
+Alternatively, the end-user can bring own CA for user authentication.
+The CO will accept the CA if the secret exists before the cluster is created. 
```

Comment:
This should be a documented procedure for the CO as it exists currently, shouldn't it?

Reply:
currently if the `my-cluster-kafka-clients-ca` Secret is already there, the CO doesn't create it. The thing is that the costumer is providing the Clients private key as well to the OpenShift cluster. That key will be used even for signing brokers certs used for client communication (server auth) other then for signing client certs (client auth) as desired by the customer.

Reply:
Yeah ... that is what we already discussed - the servers need to use different CA.

Reply:
So we'll have a clients CA as today used for signing brokers certs used for server auth on the clients communication "channel" and then we'll have another clients CA used for signing clients certs which can be provided by the user or created by the CO, is that right?

Reply:
Well, TBH the clients CA should be used for clients. We should have a separate one for servers. And as I said before, I actually think there is no need to have a different server cert for the clienttls listener and for the replication listener. Anyway, that is just about the name of the CA which should be used. Not that important for the proposal IMHO.